### PR TITLE
Make JFaceResources#managerFor public

### PR DESCRIPTION
**Pull request by vogella**

### CODE REVIEW

**Comment by iloveeclipse** [comment]
Ideally every new API should have a test added.

File: tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/resources/JFaceResourcesTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import org.junit.Test;
+
+public class JFaceResourcesTest {
+
+	@Test
+	public void ensureLocalResourceManagerCanBeCreated() {
+		Display d = Display.getDefault();
+		Shell shell = new Shell(d);
+		Composite composite = new Composite(shell, SWT.NONE);
+		LocalResourceManager localResourceManager = JFaceResources.managerFor(composite);
```
Comment (HannesWell) [inline tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/resources/JFaceResourcesTest.java:33]: Could we check here if `localResourceManager.getDevice()` returns the display `d`?
Just to have any kind of validation.

File: bundles/org.eclipse.jface/src/org/eclipse/jface/resource/JFaceResources.java [hunk 2]
```diff
@@ -597,8 +597,9 @@ public static void setFontRegistry(FontRegistry registry) {
 	 *
 	 * @param owner control whose disposal will trigger cleanup of everything in the
 	 *              registry.
+	 * @return LocalResourceManager
 	 */
-	static LocalResourceManager managerFor(Control owner) {
+	static public LocalResourceManager managerFor(Control owner) {
```
Comment (HannesWell) [inline bundles/org.eclipse.jface/src/org/eclipse/jface/resource/JFaceResources.java:603]: The usual convention is to mention the visibility first.
```suggestion
	public static LocalResourceManager managerFor(Control owner) {
```

File: tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/resources/JFaceResourcesTest.java (hunk 1 above)
Comment (szarnekow) [reply tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/resources/JFaceResourcesTest.java:33]: We should probably dispose the shell, too. And check that the resource manager is disposed afterwards?

**Comment by szarnekow** [comment]
Since the original ticket was motivated by repeated code, it would be great if the new API was used throughout the codebase (local to this repository).

**Comment by vogella** [comment]
> Since the original ticket was motivated by repeated code, it would be great if the new API was used throughout the codebase (local to this repository).

I agree, could be a good follow-up PR.

**Comment by vogella** [comment]
> Ideally every new API should have a test added.

Done

**Comment by vogella** [comment]
Virtual test failure fails randomly, see https://github.com/eclipse-platform/eclipse.platform.ui/issues/1005

**Pull request merged**